enhancements/okd-on-fedora-coreos.md

```diff
@@ -0,0 +1,76 @@
+---
+title: OKD on Fedora CoreOS
+authors:
+  - "@LorbusChris"
+reviewers:
+  - "@smarterclayton"
+  - "@miabbott"
+  - "@ashcrow"
+  - "@runcom"
+  - "@darkmuggle"
+  - "@vrutkovs"
+approvers:
+creation-date: 2019-10-21
+last-updated: 2019-10-21
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:  
+replaces:
+superseded-by:
+---
+
+# OKD-on-Fedora-CoreOS
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+ 1. This proposal potentially requires addition of non-Koji-built RPMs from OpenShift's Prow CI instance to Fedora CoreOS composes. Can we do this? If yes, how do we do this?
+ 2. What is the specific list of operators that will go into OKD-on-FCoS?
+
+ Please see the Graduation Criteria section for more context on both open questions.
+
+## Summary
+
+A proposal to create components of an OKD variant that runs atop Fedora CoreOS and to use OpenShift's Prow instance to do so.
+
+As part of the version 4 effort, the OKD working group has agreed to target Fedora CoreOS (FCoS) as the primary base operating system for builds of version 4 of the OKD Community Distribution that is powered by the OpenShift codebase.
+
+This document proposes to incrementally add Prow CI jobs to create Fedora-based artifacts for use in OKD version 4.
+
+## Motivation
+
+Delivering continuous builds of version 4 of the OKD community variant, based on Fedora CoreOS. 
+
+### Goals
+
+The broader goal is to create automated builds of all components needed for the OKD-on-FCoS effort, with a preference on running jobs in Fedora containers, and exclusively using RPMs from official Fedora repositories for Fedora CoreOS composes. Container artifacts from UBI based CI builds for OCP may be re-used, however those artifacts that come from RHEL container based builds may not be re-used to build OKD.
+
+### Non-Goals
+
+The migration to Ignition Spec v3 is not part of this proposal.
+
+## Proposal
+
+In order to create automated builds of OKD-on-FCoS, OpenShift's Prow instance shall be used to run jobs to build all required components/artifacts in Fedora containers. The definitions of these jobs will live side by side with the already existing definitions for OCP jobs in the `openshift/release` repository, on the respective operators' repos. Operators that will not go into OKD can be disregarded from this effort.
```

Comment:
Hrm, I don't really want to do Fedora for everything, but instead UBI (as we say below)

Reply:
Maybe I just misread this section

Reply:
At least for RPM builds, I really want to avoid any potential problems caused by skew between UBI/RHEL and Fedora as buildsystem. UBI everywhere else is fine with me.

Reply:
I clarified the passage

Reply:
`with a preference on running RPM build jobs in Fedora containers, and exclusively using RPMs from official Fedora repositories for Fedora CoreOS composes`

Its important to mention that binaries running on the host (hyperkube, crio, machine-config-daemon) should be built on Fedora (preferably via RPMs) to be compatible with i.e. glibc breaks